spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java

```diff
@@ -164,10 +169,23 @@ public BinderAwareChannelResolver binderAwareChannelResolver(BindingService bind
 	}
 
 	@Bean
-	@ConditionalOnProperty("spring.cloud.stream.bindings." + ERROR_CHANNEL_NAME + ".destination")
-	public SingleBindingTargetBindable<MessageChannel> errorChannelBindable(
+	@ConditionalOnProperty("spring.cloud.stream.bindings." + ERROR_KEY_NAME + ".destination")
+	public MessageChannel errorBridgeChannel(
 			@Qualifier(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME) PublishSubscribeChannel errorChannel) {
```

Comment:
There is no guarantee that end-user doesn't override global `errorChannel` with its own implementation.
We may insist to restrict it into the `SubscribeChannel` to simplify our life, but that really should not be just `PublishSubscribeChannel`.

We may reconsider this in the version `2.0` when SI 5.0 is available with its `FluxMessageChannel` though